lib/api/controllers/adminController.js

```diff
@@ -54,6 +58,15 @@ class AdminController {
         return this.kuzzle.internalEngine.bootstrap.all();
       });
 
+    if (refresh === 'wait_for') {
+      return promise.then(() => {
+        delete _locks.resetKuzzleData;
```

Comment:
The lock is removed only if the promise chain succeeds. If it fails, the resource remains locked forever.